### PR TITLE
Add Arango Pregel algorithms during network creation

### PR DESCRIPTION
**Pull request by curtislisle**
### Does this PR close any open issues?
No
### Give a longer description of what this PR addresses and why it's needed
When new networks are created, pagerank and community detection algorithms are executed inside Arango and new attributes are added to the node table(s).  Tested with the default initialized databases and some uploaded ones. 

### Are there any additional TODOs before this PR is ready to go?
TODOs:
- [ ] Update relevant documentation
- [ ] ...

### CODE REVIEW

File: multinet/api/models/network.py [hunk 1]
```diff
@@ -79,6 +82,102 @@ def create_with_edge_definition(
...
+                    store=True,
+                    #maxx_gss=100,
+                    #thread_count=1,
+                    async_mode=False,
+                    result_field='_pagerank'
+                )
+                # a conflict writing results were noticed when multiple jobs finish about the same time, so 
+                # wait after each job before starting the next one
+
+                while (arango_db.pregel.job(pagerank_job_id)['state'] == 'running'):
```
Comment (JackWilb) [inline multinet/api/models/network.py:106]: Looks like there is a state for `storing` the data back to the db

Check here for the possible states: https://www.arangodb.com/docs/stable/http/pregel.html#get-pregel-job-execution-status

You will probably want to update all the instances that we are waiting for a job to finish

**Review by JackWilb — COMMENTED** [review]
It's awesome that we can do this in arango; it makes the whole thing much more simple. I left one comment on waiting for potential states, and I believe we're missing the bind var changes. Maybe they just need to be pushed.

File: multinet/api/models/network.py (hunk 1 above)
Comment (curtislisle) [reply multinet/api/models/network.py:106]: Thanks for noticing the `storing` state.  I am adding this into the code now and will report during our call.  Correct, that I just haven't pushed the version with the bound variables yet.  Thanks for testing this.

Comment (curtislisle) [reply multinet/api/models/network.py]: I just pushed a commit to add bound variables and add a test for `storing` data after the job is finished.  I'll look into whether  arango supports a write lock to allow the jobs to run in parallel and then write sequentially.   This passes all the demo datasets.  

In our meeting today, it was suggested to move this code into a task executed after a network is generated.

Comment (curtislisle) [reply multinet/api/models/network.py]: TODO:  I see why we need to separate this code into a separate task.  It also needs to be executed in the upload task in csv.py when I network is created by the UI.

**Comment by curtislisle** [comment]
These changes (Pregel and AQL) have been added to a new  `pregel` branch of this repo.  I am withdrawing this PR.  After refactoring by Kitware, another PR will be made based on the new branch.

**Pull request closed without merging**